util/secret/certificates.go

```diff
@@ -456,3 +521,32 @@ func newSelfSignedCACert(key *rsa.PrivateKey) (*x509.Certificate, error) {
 	c, err := x509.ParseCertificate(b)
 	return c, errors.WithStack(err)
 }
+
+func newClientCert(cn string, caCert *x509.Certificate, key *rsa.PrivateKey, caKey *rsa.PrivateKey) (*x509.Certificate, error) {
+	cfg := certs.Config{
+		CommonName: cn,
+	}
+
+	now := time.Now().UTC()
+
+	tmpl := x509.Certificate{
+		SerialNumber: new(big.Int).SetInt64(0),
+		Subject: pkix.Name{
+			CommonName:   cfg.CommonName,
+			Organization: cfg.Organization,
+		},
+		NotBefore:   now.Add(time.Minute * -5),
+		NotAfter:    now.Add(time.Hour * 24 * 365 * 10), // 10 years
```

Comment:
Worth defining as a constant somewhere?

Off-topic thought: I don't think we support user-defined expiration in general, but it might be a nice feature

Reply:
I see we define this in line here: https://github.com/kubernetes-sigs/cluster-api/blob/f248367309a112b021c95bd8747b17200739c18a/util/secret/certificates.go#L443

Reply:
+1 for consts

Reply:
+100